src/CashuWallet.ts

```diff
@@ -88,25 +88,44 @@ class CashuWallet {
 	}
 
 	async receive(encodedToken: string): Promise<Array<Proof>> {
-		const { token } = getDecodedToken(encodedToken);
-		const amount = token[0]?.proofs?.reduce((total, curr) => total + curr.amount, 0);
-		const { payload, amount1BlindedMessages, amount2BlindedMessages } =
-			await this.createSplitPayload(0, amount, token[0]?.proofs);
-		const { fst, snd } = await this.mint.split(payload);
-		const proofs1 = dhke.constructProofs(
-			fst,
-			amount1BlindedMessages.rs,
-			amount1BlindedMessages.secrets,
-			this.keys
-		);
-		const proofs2 = dhke.constructProofs(
-			snd,
-			amount2BlindedMessages.rs,
-			amount2BlindedMessages.secrets,
-			this.keys
-		);
-		const newProofs = [...proofs1, ...proofs2];
-		return newProofs;
+		const { token: tokens } = getDecodedToken(encodedToken);
+		const proofs: Proof[] = [];
+		const mintKeys = new Map<string, MintKeys>([[this.mint.mintUrl, this.keys]]);
+		for (const token of tokens) {
+			if (!token?.proofs || !token?.mint) {
+				continue;
+			}
+			try {
+				const keys = mintKeys.get(token.mint) || (await new CashuMint(token.mint).getKeys());
+				const amount = token.proofs.reduce((total, curr) => total + curr.amount, 0);
+				const { payload, amount1BlindedMessages, amount2BlindedMessages } =
+					await this.createSplitPayload(0, amount, token.proofs);
+				const { fst, snd } = await CashuMint.split(token.mint, payload);
+				const proofs1 = dhke.constructProofs(
+					fst,
+					amount1BlindedMessages.rs,
+					amount1BlindedMessages.secrets,
+					keys
+				);
+				const proofs2 = dhke.constructProofs(
+					snd,
+					amount2BlindedMessages.rs,
+					amount2BlindedMessages.secrets,
+					keys
+				);
+				proofs.push(...proofs1, ...proofs2);
+				if (!mintKeys.has(token.mint)) {
+					mintKeys.set(token.mint, keys);
+				}
+			} catch (error) {
+				// I'm not sure if this is the best way to handle errors
+				if (proofs.length) {
+					return proofs;
+				}
+				throw error;
+			}
+		}
+		return proofs;
```

Comment:
It's a good question. Since we are dealing with multiple entpoints, some stuff might go through and some stuff might not. It is a pretty clean solution, my only issue with it is that, we will get some proofs back, but those that couldn't be redeemed (maybe a mint is down or something) are now in an invalid token-set (because those that were redeemed are now invalid) , so we will get another error if we try to redeem again for the tokens that had problems. 

Maybe we could return the failed tokens too, so we can only check those once more? What do you think?

something like:
```suggestion
			} catch (error) {
				// I'm not sure if this is the best way to handle errors
				console.error(error)
				tokensWithErrors.push(token)
			}
		}
		return {proofs, tokensWithErrors};
```

if we have any tokensWithError, we can handle them outside of the lib